code/modules/surgery/bodyparts/_bodyparts.dm

```diff
@@ -636,9 +640,25 @@
 		if(needs_update_disabled)
 			update_disabled()
 
+	update_bodypart_traits(new_owner)
 	refresh_bleed_rate()
 	return old_owner
 
+/obj/item/bodypart/proc/update_bodypart_traits()
+	for(var/trait in bodypart_traits)
+		ADD_TRAIT(owner, trait, bodypart_trait_source)
+
+/obj/item/bodypart/proc/add_bodypart_trait(trait)
+	bodypart_traits += trait
+	update_bodypart_traits()
+
+/obj/item/bodypart/proc/remove_bodypart_trait(trait)
+	bodypart_traits -= trait
+	REMOVE_TRAIT(owner, trait, bodypart_trait_source)
+
+/obj/item/bodypart/proc/on_removal()
+	for(var/trait in bodypart_traits)
+		REMOVE_TRAIT(owner, trait, bodypart_trait_source)
```

Comment:
this is super unnecessary, just use `try_attach_limb()` and `drop_limb()` for this stuff. This is just bloat.

Reply:
If you say so man, it's meant to be for tools used in further functionality.

Reply:
> If you say so man, it's meant to be for tools used in further functionality. What if someone wanted to add or remove a trait from a limb on the fly?

imho that behavior shouldnt even be allowed. I don't see a design space where thats necessary.

And even then, `on_removal()` is still a proc that doesnt need to exist, we have `drop_limb()` overrides for that currently.

Reply:
You are correct about that second part for sure.